src/algebra/group/basic.lean

```diff
@@ -39,6 +39,11 @@ end associative
 section semigroup
 variables {α : Type*}
 
+@[to_additive]
+instance : semigroup empty :=
+{ mul := λ x y, by cases x,
+  mul_assoc := λ x y z, by cases x }
```

Comment:
Should `pempty.{u}` get an instance too? How about instance of `has_sub`, `has_neg` etc as well?

Reply:
I have to admit that I don't know what `pempty.{u}` is. I don't mind adding `has_sub`, `has_neg` as well, but at the moment I don't see a purpose as you would usually use `has_sub` where you have some neutral element. If others have an opinion, feel free to comment. These instances seem not to help with this PR though.

Reply:
I guess what I'm suggesting is a `algebra/pempty_instances` file to go alongside `algebra/punit_instances`. `empty` is a shorthand for `pempty.{1}` as far as I remember.

Reply:
That sounds reasonable. But maybe one file is enough and call it something like subsingleton_instances?

Reply:
I'd find that name confusing, because it suggests the instances are all `instance : subsingleton`. I don't think there's any real cost associated with creating the new file.

Reply:
Ok, I'll add the file with the semigroup instance of pempty, but I'm not convinced that has_sub and has_neg should be there. I don't really regard them as algebraic structures in their own right. But of course it is no problem at all to add them if you convince me otherwise. What would be the purpose?

Reply:
I don't feel strongly about has_neg and has_sub, fine to leave those out.